### ISSUE TITLE
Add Invoke Command Suppress response support 

### ISSUE THREAD

**Issue by yunhanw-google** [project-chip/connectedhomeip#8043]
#### Problem
there is spec conflict regarding suppress response flag in https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/3536, after the above issue is resolved, we would decide how to add Invoke Command Suppress response support.

**Comment by mrjerryjohns** [comment]
Was resolved a while back.

**Comment by bzbarsky-apple** [comment]
No, this is not supported.  We parse it, but do absolutely nothing with it.

**Comment by Rajashreekalmane** [comment]
@bzbarsky-apple @mrjerryjohns In Spec mentioned suppress response is mandatory for invokerequest and write request message 
![image](https://user-images.githubusercontent.com/95841834/171809801-42633417-8497-4164-bc10-ccd88805ef16.png)
Could you please let me know how to check whether DUT  suppress its response when TH sends Invoke/write request with suppress response parameter set to bool.

**Comment by bzbarsky-apple** [comment]
> Could you please let me know how to check whether DUT suppress its response

Does the TH get a response?  If so, the DUT did not suppress it.

**Comment by sankarselvam** [comment]
@bzbarsky-apple From my observation, DUT is not suppressing the response. 
@Rajashreekalmane , Can you please attach the logs?

**Comment by bzbarsky-apple** [comment]
> @bzbarsky-apple From my observation, DUT is not suppressing the response.

Right, that's why this issue is open!

>  @Rajashreekalmane , Can you please attach the logs?

There is no need to attach logs here.

**Comment by Rajashreekalmane** [comment]
@bzbarsky-apple what should I do in this case, Is this cert-blocker  or V1.0 out of scope issue.

**Comment by bzbarsky-apple** [comment]
So far it's being treated as out of scope.

**Comment by vivien-apple** [comment]
Just to make it clear, the `chip-tool` syntax for such a command is:
```
$ ./out/debug/standalone/chip-tool onoff toggle 0x12344321 1 --suppressResponse true
```

It does not work as of today because the stack does not supports suppress response. It can be observed that the value is set properly in the `InvokeRequestMessage` that is logged.

**Comment by VaishaliAvhale** [comment]
With the above MR is this issue resolved?

**Comment by bzbarsky-apple** [comment]
@VaishaliAvhale I don't know what you mean by "above MR".  This issue is not resolved.

**Comment by VaishaliAvhale** [comment]
I was talking about https://github.com/project-chip/connectedhomeip/pull/27024 this MR.
I wanted to know if the issue has been resolved using the above MR.

**Comment by VaishaliAvhale** [comment]
@bzbarsky-apple Please ignore the comments. Sorry for the confusion.

**Comment by j-ororke** [comment]
Adding a comment here: from my recent work being done on implementing IDM_3_2 python3 test module here: [41066](https://github.com/project-chip/connectedhomeip/pull/41066).
I noticed that test step 4 had this above issue noted for why it was currently not being run for the yaml script version of this test as well as being marked as "out of scope":

<img width="449" height="65" alt="Image" src="https://github.com/user-attachments/assets/7796433e-da14-4303-bfcd-7dd3ef9db1a6" />

However, @cecille had found that a good majority of the plumbing here was already done but was just missing the exposure in the python test framework. So under her guidance I commenced working on the final steps to complete the plumbing and exposing it to use in the IDM_3_2 test for test step 4, during this I found that the server side C++ WriteHandler's HandleWriteRequestMessage() method was not checking for the SuppressResponse flag before sending WriteResponse messages back to the Controller regardless of if the SuppressResponse was set to True or not. This caused the SuppressResponse flag to go unchecked for any write operations done, as the DUT would always respond regardless of whether the SuppressResponse flag was set to True.

I added onto the if check below in the C++ WriteHandler's HandleWriteRequestMessage() method and rebuilt the all-clusters app with this change made, we are no longer receiving the responses back to write attribute operations for the IDM_3_2 test during test step 4 afterwards, also when we check and make sure that the attribute actually got written too, it has the updated value from the write operation done in test step 4:

<img width="972" height="46" alt="Image" src="https://github.com/user-attachments/assets/e0c82a44-68f6-4fc1-9d0c-79ed78d01e50" />
